contracts/src/v0.8/functions/tests/v1_X/FunctionsBilling.t.sol

```diff
@@ -149,24 +220,147 @@ contract FunctionsBilling__StartBilling {
   // TODO: make contract internal function helper
 }
 
-/// @notice #_computeRequestId
-contract FunctionsBilling__ComputeRequestId {
-  // TODO: make contract internal function helper
-}
-
 /// @notice #_fulfillAndBill
 contract FunctionsBilling__FulfillAndBill {
   // TODO: make contract internal function helper
 }
 
 /// @notice #deleteCommitment
-contract FunctionsBilling_DeleteCommitment {
+contract FunctionsBilling_DeleteCommitment is FunctionsClientRequestSetup {
+  function test_DeleteCommitment_RevertIfNotRouter() public {
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
 
+    vm.expectRevert(Routable.OnlyCallableByRouter.selector);
+    s_functionsCoordinator.deleteCommitment(s_requests[1].requestId);
+  }
+
+  event CommitmentDeleted(bytes32 requestId);
+
+  function test_DeleteCommitment_Success() public {
+    // Send as Router
+    vm.stopPrank();
+    vm.startPrank(address(s_functionsRouter));
+
+    // topic0 (function signature, always checked), NOT topic1 (false), NOT topic2 (false), NOT topic3 (false), and data (true).
+    bool checkTopic1 = false;
+    bool checkTopic2 = false;
+    bool checkTopic3 = false;
+    bool checkData = true;
+    vm.expectEmit(checkTopic1, checkTopic2, checkTopic3, checkData);
+    emit CommitmentDeleted(s_requests[1].requestId);
+
+    s_functionsCoordinator.deleteCommitment(s_requests[1].requestId);
+  }
 }
 
 /// @notice #oracleWithdraw
-contract FunctionsBilling_OracleWithdraw {
+contract FunctionsBilling_OracleWithdraw is FunctionsMultipleFulfillmentsSetup {
+  function test_OracleWithdraw_RevertWithNoBalance() public {
+    uint256 transmitter1BalanceBefore = s_linkToken.balanceOf(NOP_TRANSMITTER_ADDRESS_1);
+    assertEq(transmitter1BalanceBefore, 0);
+    uint256 transmitter2BalanceBefore = s_linkToken.balanceOf(NOP_TRANSMITTER_ADDRESS_2);
+    assertEq(transmitter2BalanceBefore, 0);
+    uint256 transmitter3BalanceBefore = s_linkToken.balanceOf(NOP_TRANSMITTER_ADDRESS_3);
+    assertEq(transmitter3BalanceBefore, 0);
+    uint256 transmitter4BalanceBefore = s_linkToken.balanceOf(NOP_TRANSMITTER_ADDRESS_4);
+    assertEq(transmitter4BalanceBefore, 0);
+
+    // Send as stranger, which has no balance
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    vm.expectRevert(FunctionsSubscriptions.InvalidCalldata.selector);
+
+    // Attempt to withdraw with no amount, which would withdraw the full balance
+    s_functionsCoordinator.oracleWithdraw(STRANGER_ADDRESS, 0);
+
+    uint96 expectedTransmitterBalance = 0;
+
+    uint256 transmitter1BalanceAfter = s_linkToken.balanceOf(NOP_TRANSMITTER_ADDRESS_1);
```

Comment:
Can we turn this repeated code into a helper method?

Reply:
I'll add helpers to get all transmitter balances, how about that?
I'd rather keep the assertions within the test for now